### PR TITLE
Security hub overview

### PR DESCRIPTION
**Pull request by njgarratt**
In a comment on the pull request, include the following information and check off the list.

 - [x] Mini bio
 Nick Garratt is a Principal Architect in the Rackspace Managed Public Clouds Product Architecture team. Nick has over 20 years experience in the IT Sector where he has occupied diverse roles spanning Hardware, Networking, Unix and Unix like Operating Systems, Software Development and  Cloud Computing, many of these being customer facing roles.
 - [x] Desired date of publishing
  2018-12-12 (today)
 - [x] Social media contact info
https://www.linkedin.com/in/nicholasgarratt
 - [x] Redaction: (yes|no)
yes
 - [x] Comments: (yes|no)
yes

@catlook Please review.

### CODE REVIEW

**Comment by rackernexus** [comment]
If you want to double-check your formatting, check [the preview](https://staging.developer.rackspace.com/build-00b0666157/).

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-11d3b888e0/)

**Comment by njgarratt** [comment]
Hello InfoDev, @catlook ,

This looks good to go from my perspective.

Thanks,

Nick

**Comment by catlook** [comment]
I'll review this and let you know!

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/build-e0ab8bf99d/). :bow:

**Comment by njgarratt** [comment]
Thank you!

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+<!-- more -->
+
+## Security Hub - why is this significant?
+
+One of the notable developments included in this announcement is the creation by AWS and adoption by AWS and select AWS partners of a **standard format** for security events called the ['AWS Security Finding' format](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-findings-format.html). This common format is a key enabler of the aggregation of 'Findings' into Security Hub as it pushes the responsibility for conformance onto the findings emitter rather than requiring the aggregator to build and maintain multiple parsers for all findings sources. The AWS services GuardDuty, Macie and Inspector, if configured, will automatically have their findings aggregated in SecurityHub once it is enabled.
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: ... Macie, and Inspector, ...

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+One of the notable developments included in this announcement is the creation by AWS and adoption by AWS and select AWS partners of a **standard format** for security events called the ['AWS Security Finding' format](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-findings-format.html). This common format is a key enabler of the aggregation of 'Findings' into Security Hub as it pushes the responsibility for conformance onto the findings emitter rather than requiring the aggregator to build and maintain multiple parsers for all findings sources. The AWS services GuardDuty, Macie and Inspector, if configured, will automatically have their findings aggregated in SecurityHub once it is enabled. 
+ 
+Given that many of the security partners involved compete with one another to some extent or offer SOC services providing the black box magic of event ingestion and correlation into incidents, one can imagine that the incentive to standardise on a format which could level the playing field just a little bit must not have been great. The inexorable change of the market and the dominant agent of that change together clearly made a compelling case!
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: [suggestion: spell out SOC - what is it?]

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+Under the 'Standards' menu, AWS provides the first of its automated compliance checks – 'CIS AWS Foundations' -  with 43 automated rules as of 11th December 2018. 
+ 
+## A brisk walk-through
+
+Let's take a brief look in practice at how these various concepts and components work together. We'll be working with a single test account rather than trying to demonstrate master/member.
+
+* We've enabled GuardDuty for our test account to provide us with some findings to work with.
+* In the ** AWS Console -> GuardDuty -> Settings ** we've clicked 'Generate Sample Findings' which has generated a spread of 45 findings.
+* Since GuardDuty findings are automatically aggregated in SecurityHub, after a short delay, these same 45 findings are available for us to manipulate in SecurityHub.
+* Today we are concerned about resource theft for BitCoin mining, let's see what GuardDuty has identified for us in this regard:
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: Today, we are  ...
... mining, so let's see ...

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+```bash
+aws securityhub create-insight --name "bitcoin miner"  --filter '{"ResourceType": [ { "Comparison": "CONTAINS", "Value": "AwsEc2Instance" } ], "Type": [{"Comparison": "CONTAINS", "Value": "TTPs/Command and Control/CryptoCurrency:EC2-BitcoinTool.B!DNS"}] }' --group-by AwsAccountId
+{
+    "InsightArn": "arn:aws:securityhub:eu-west-1:9876123456789:insight/9876123456789/custom/45be0f15-d947-4bf7-8c27-83eef7487141"
+}
+
+```
+
+* There is no aws CLI option to create an `action` just now, so we'll need to got to the ** AWS Console -> SecurityHub -> Settings -> Custom actions. ** Click on 'Create custom action'
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: ... to go to the  ...

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+![Custom Action]({% asset_path 2018-12-12-aws-security-hub-overview/custom_action.png %})
+
+* It's now time to create our CloudWatch Events rule:
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: ... our **CloudWatch Events** rule: ...

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+}
+```
+
+* For our `rule` `target`, we'll make use of an SNS topic which is already in use for other security notifications:
+
+```bash
+aws events put-targets --rule SecHub --targets "Id"="1","Arn"="arn:aws:sns:eu-west-1:9876123456789:security_events"
+```
+
+our subscription to this topic looks like this:
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: Our subscription ...

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+* In the ** AWS Console -> GuardDuty -> Settings ** we've clicked 'Generate Sample Findings' which has generated a spread of 45 findings.
+* Since GuardDuty findings are automatically aggregated in SecurityHub, after a short delay, these same 45 findings are available for us to manipulate in SecurityHub.
+* Today we are concerned about resource theft for BitCoin mining, let's see what GuardDuty has identified for us in this regard:
+
+
+``` bash
+aws securityhub get-findings --filter '{"ResourceType": [ { "Comparison": "CONTAINS", "Value": "AwsEc2Instance" } ], "Type": [{"Comparison": "CONTAINS", "Value": "TTPs/Command and Control/CryptoCurrency:EC2-BitcoinTool.B!DNS"}] }'
+```
+
+and indeed we do have a match output in the AWS Security Finding format:
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: And, indeed, we  ...
... the **AWS Security Finding format**: ...

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+        "TopicArn": "arn:aws:sns:eu-west-1:9876123456789:security_events"
+    }
+}
+```
+
+* So far, so good! Time for action...In the ** AWS Console -> Security Hub -> Insights ** panel, we find the `insight` which we created earlier:
+
+![Insight]({% asset_path 2018-12-12-aws-security-hub-overview/insight.png %})
+
+* Clicking through on that `insight` we see the `finding` we previously turned up in our seach. We highlight this and select the `action` we created earlier from the dropdown. We are notified in a green across the top of the panel that this operation has been successful.
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: ... that `insight`, we see ...

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+![Action]({% asset_path 2018-12-12-aws-security-hub-overview/action.png %})
+
+* While we wait for the email notification to arrive we can validate that CloudWatch Events has seen and processed the event triggered by our `action` by visiting ** AWS Console -> CloudWatch -> Events -> Rules ** and click on our "SecHub" rule:
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: ... to arrive, we can ...
... and clicking on our **SecHub** rule:  ...

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+![Action]({% asset_path 2018-12-12-aws-security-hub-overview/cw_events.png %})
+
+If we view the metrics for this rule, we should should see `TriggeredRules` reporting a value of `1`. Before long, however, an email will appear in our Inbox with the full raw contents of the `finding`. 
+
+## Where to from here?
+
+As a PoC, this is interesting, but this is merely a start. CloudWatch Events provides a broad selection of targets which enable a high degree of versatility, whether you want to push events to another system, generate a ticket in order to engage human specialists, or automate remediation. What can you accomplish? Sign up for the Security Hub preview and share your thoughts and experiences below.
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: ... targets, which ...

File: _posts/2018-12-12-aws-security-hub-overview.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+
+Something to watch is the extent to which the AWS Security Finding Format takes on a life of its own outside of AWS. Having been compelled to conform to a standard, does this development serve as a catalyst for further cooperation between vendors or the inspiration behind yet to be conceived OSS projects? There does not appear to have been much movement on Security Device Event Exchange and the Cisco extension [CICEE](https://www.cisco.com/c/en/us/td/docs/security/ips/specs/CIDEE_Specification.html?dtid=osscdc000283) in recent years, just perhaps this is the impetus which has been lacking.
```
Comment (catlook) [inline _posts/2018-12-12-aws-security-hub-overview.md]: ... the **AWS Security Finding format** takes ...

File: _posts/2018-12-12-aws-security-hub-overview.md (hunk 1 above)
Comment (njgarratt) [reply _posts/2018-12-12-aws-security-hub-overview.md]: oxford commas!

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/build-175f2af0c2/). :bow:

**Pull request merged**